### DISCUSSION TITLE
Is there a .h file or method that could parse .astc like stbi parsing .png?

### DISCUSSION THREAD

**Discussion by dreakWang** [KhronosGroup/KTX-Software#514] [Q&A] [answered]
Hi, I am looking for a method could directly parse .astc file, any suggestions? Thanks!

Besides, from the 
https://github.com/ARM-software/ktx-creator
it says "toktx --encode astc --astc_blk_d 6x6 --astc_mode ldr --genmipmap output.ktx input[s].png" could encode png to astc and wrap it to the ktx, but this command doesn't work and doesn't give me any error message, and I don't see any --encode astc message in the toktx --help. Is it deprecated? Any other command could transfer .png to .ktx? Thanks!

**Comment by MarkCallow** [comment, marked as answer]
I don't know what a .astc file is so I can't help with that part of your question.

The ASTC encoder is included in the top-of-tree `toktx` which has not yet been released as we are still working on the normal mode feature. Presumably you are using the version from the Releases pages which is why you aren't seeing ASTC in the help. If you clone and build the software yourself, you will be able to create KTX files with ASTC-encoded payloads.

**Reply by dreakWang** [reply]
Thanks for reply, the .astc file is created by .png or .jpg ASTC encoder encoding.
